src/devices/src/virtio/net/device.rs

```diff
@@ -161,13 +188,20 @@ impl Net {
             | 1 << VIRTIO_F_VERSION_1
             | 1 << VIRTIO_RING_F_EVENT_IDX;
 
+        let mac_addr: Option<MacAddr>;
         let mut config_space = ConfigSpace::default();
-        if let Some(mac) = guest_mac {
-            config_space.guest_mac.copy_from_slice(mac.get_bytes());
-            // When this feature isn't available, the driver generates a random MAC address.
-            // Otherwise, it should attempt to read the device MAC address from the config space.
-            avail_features |= 1 << VIRTIO_NET_F_MAC;
+        // When this feature isn't available, the driver generates a random MAC address.
+        // Otherwise, it should attempt to read the device MAC address from the config space.
+        avail_features |= 1 << VIRTIO_NET_F_MAC;
+        if guest_mac.is_some() {
+            mac_addr = guest_mac.copied();
+        } else {
+            // When the MAC address is not specified explicitely, generate one and assign it.
+            mac_addr = generate_mac_address();
         }
+        config_space
+            .guest_mac
+            .copy_from_slice(mac_addr.unwrap().get_bytes());
```

Comment:
Is this unwrap safe?

Reply:
Yes, I believe so. As `mac_addr` is assigned with `guest_mac` or from the result of `generate_mac_address()` which will both be valid